source/ios/AdaptiveCards/ADCIOSVisualizer/ADCIOSVisualizer/CustomTextBlockRenderer.mm

```diff
@@ -0,0 +1,109 @@
+    NSMutableString *fontFamilyName = [[NSMutableString alloc] initWithString:@"'"];
+    for(NSUInteger index = 0; index < [acoConfig.fontFamilyNames count] - 1; ++index){
+        [fontFamilyName appendString:acoConfig.fontFamilyNames[index]];
+        [fontFamilyName appendString:@"', '"];
+    }
+    [fontFamilyName appendString:acoConfig.fontFamilyNames[[acoConfig.fontFamilyNames count] - 1]];
+    [fontFamilyName appendString:@"'"];
+
+    // Font and text size are applied as CSS style by appending it to the html string
+    parsedString = [parsedString stringByAppendingString:[NSString stringWithFormat:@"<style>body{font-family: %@; font-size:%dpx; font-weight: %d;}</style>",
+                                                          fontFamilyName,
+                                                          [acoConfig getTextBlockTextSize:textConfigForBlock.size],
+                                                          [acoConfig getTextBlockFontWeight:textConfigForBlock.weight]]];
+    // Convert html string to NSMutableAttributedString, NSAttributedString knows how to apply html tags
+    NSData *htmlData = [parsedString dataUsingEncoding:NSUTF16StringEncoding];
+    NSDictionary *options = @{NSDocumentTypeDocumentAttribute: NSHTMLTextDocumentType};
+
+    // Initializing NSMutableAttributedString for HTML rendering is very slow
+    NSMutableAttributedString *content = [[NSMutableAttributedString alloc] initWithData:htmlData options:options documentAttributes:nil error:nil];
+
+    UILabel *lab = [[UILabel alloc] init]; // generate key for text map from TextBlock element's id
+    // Set paragraph style such as line break mode and alignment
+    NSMutableParagraphStyle *paragraphStyle = [[NSMutableParagraphStyle alloc] init];
+    paragraphStyle.lineBreakMode = textConfigForBlock.wrap ? NSLineBreakByWordWrapping:NSLineBreakByTruncatingTail;
+    paragraphStyle.alignment = [ACOHostConfig getTextBlockAlignment:textBlockElement->GetHorizontalAlignment()];
+
+    // Obtain text color to apply to the attributed string
+    ACRContainerStyle style = viewGroup.style;
+    ColorsConfig &colorConfig = (style == ACREmphasis)? [acoConfig getHostConfig]->containerStyles.emphasisPalette.foregroundColors:
+                                                                           [acoConfig getHostConfig]->containerStyles.defaultPalette.foregroundColors;
+    // Add paragraph style, text color, text weight as attributes to a NSMutableAttributedString, content.
+    [content addAttributes:@{NSParagraphStyleAttributeName:paragraphStyle, NSForegroundColorAttributeName:[ACOHostConfig getTextBlockColor:textConfigForBlock.color colorsConfig:colorConfig subtleOption:textConfigForBlock.isSubtle],} range:NSMakeRange(0, content.length)];
+    lab.attributedText = content;
+
+    lab.numberOfLines = int(textBlockElement->GetMaxLines());
+    if(!lab.numberOfLines and !textBlockElement->GetWrap()){
```

Comment:
>and [](start = 26, length = 3)

`&&` is really preferable to `and`. It's not wrong, but the alternative operator names are hardly ever used.